contrib/platform/test/com/sun/jna/platform/win32/NamedPipeTest.java

```diff
@@ -0,0 +1,181 @@
+/* Copyright (c) 2007 Timothy Wall, All Rights Reserved
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+package com.sun.jna.platform.win32;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import org.junit.Test;
+
+import com.sun.jna.platform.win32.WinDef.ULONGByReference;
+import com.sun.jna.platform.win32.WinNT.HANDLE;
+import com.sun.jna.ptr.IntByReference;
+
+/**
+ * @author lgoldstein
+ */
+public class NamedPipeTest extends AbstractWin32TestSupport {
+    public NamedPipeTest() {
+        super();
+    }
+
+    @Test
+    public void testNamedPipeServerAPI() {
+        String    pipeName="\\\\.\\pipe\\" + getCurrentTestName();
+        HANDLE    hNamedPipe=Kernel32.INSTANCE.CreateNamedPipe(pipeName,
+                WinBase.PIPE_ACCESS_DUPLEX,        // dwOpenMode
+                WinBase.PIPE_TYPE_BYTE | WinBase.PIPE_READMODE_BYTE | WinBase.PIPE_WAIT,    // dwPipeMode
+                1,    // nMaxInstances,
+                Byte.MAX_VALUE,    // nOutBufferSize,
+                Byte.MAX_VALUE,    // nInBufferSize,
+                1000,    // nDefaultTimeOut,
+                null);    // lpSecurityAttributes
+        assertCallSucceeded("CreateNamedPipe", !WinBase.INVALID_HANDLE_VALUE.equals(hNamedPipe));
+
+        // NOTE: we don't really care what the returned values are only that the call succeeds
```

Comment:
I think it does explain to whoever did NOT write the test why the values retrieved from the call are not tested.